proc_CPMG.py

```diff
@@ -5,31 +5,31 @@
 fl = figlist_var()
 logger = init_logging('info')
 for searchstr, exp_type, nodename, postproc, label_str in [
-        #('200221','CPMG_TEMPOLgel_3p0_1','deadtime=5'),
+        #('200221_CPMG_TEMPOLgel_3p0_1','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
         ('200221_CPMG_TEMPOLgel_2p9_1','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
-        #('200304','CPMG_2p6_1','deadtime=5'),
-        #('200305','CPMG_3p5_2','deadtime=5'),
-        #('200305','CPMG_3p6_2','deadtime=5'),
-        #('200305','CPMG_3p7_2','deadtime=5'),
-        #('200305','CPMG_3p7_3','deadtime=5'),
-        #('200305','CPMG_3p8_2','deadtime=5'),
-        #('200305','CPMG_3p9_2','deadtime=5'),
-        #('200305','CPMG_4p0_1','deadtime=5'),
+        #('200304_CPMG_2p6_1','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p5_2','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p6_2','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p7_2','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p7_3','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p8_2','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_3p9_2','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
+        #('200305_CPMG_4p0_1','test_equip','signal','spincore_CPMG_v1','deadtime=5'),
         ]:
     s =  find_file(searchstr, exp_type=exp_type,
             expno=nodename, postproc=postproc, lookup=postproc_dict)
-    nEchoes = s.get_prop('acq_params')['nEchoes']
     #{{{select and display coherence channel centered
     best_shift = hermitian_function_test(s)
     logger.info(strm("best shift is",best_shift))
-    s.setaxis('t2',lambda x:x-best_shift)
+    s.setaxis('t2', lambda x: x-best_shift)
     fl.next('time domain after hermitian test')
     fl.image(s)
     s.register_axis({'t2':0})
-    s = s['t2':(-0.01,0.01)]
```

Comment:
here, you're taking an arbitrary hard-coded slice in the time domain -- if you want to do that, then add the time range as part of the list of tuples above, and show what it looks like using the side by side plotting (as done elsewhere)

Reply:
this pertains to issue #20 (hopefully this works where you can click it....) where I slice out the echo as close as I can, but I must have done something wrong or misunderstood what you said... For clarity this is the third step you mention in issue #20

Reply:
so, then I think the answer is that if you want it to be "as large as you can", then it can't be hard coded.  So, actually, I think the goal here should be go deal with #21 :

- look at the comments there and note how I talk about separating a function to do the CPMG echo centering (this comment is referenced in issue #20)
- it should be pretty easy to test DOSY_T2, since only some 4 files are changed relative to alec_fitdata_rebased.
- you can review and merge #21 (the pull request previously had master set as the base base (the branch it wants to merge into), but it should actually merge with alec_fitdata_rebased, since it was created as a branch of alec_fitdata_rebased, so I fixed that)
- use the function created above at this spot in the code as well, which will solve this thread

Reply:
function added